File: AspNetMvcUnleashed/0672329980_source/Chapter08/VB/MvcApplication1/MvcApplication1/Models/ProductRepository.vb
Public Class ProductRepository
    Implements IProductRepository

    Private _entities As New ProductsDBEntities()

    Public Function ListProducts() As IEnumerable(Of Product) Implements IProductRepository.ListProducts
        Return _entities.ProductSet.ToList()
    End Function

    Public Sub CreateProduct(ByVal productToCreate As Product) Implements IProductRepository.CreateProduct
        _entities.AddToProductSet(productToCreate)
        _entities.SaveChanges()
    End Sub

End Class

Public Interface IProductRepository
    Function ListProducts() As IEnumerable(Of Product)
    Sub CreateProduct(ByVal productToCreate As Product)
End Interface

File: AspNetMvcUnleashed/0672329980_source/Chapter08/CS/MvcApplication1/MvcApplication1/Models/ProductRepository.cs
using System.Collections.Generic;
using System.Linq;

namespace MvcApplication1.Models
{
    public class ProductRepository : IProductRepository
    {
        private ProductsDBEntities _entities = new ProductsDBEntities();

        public IEnumerable<Product> ListProducts()
        {
            return _entities.ProductSet.ToList();
        }

        public void CreateProduct(Product productToCreate)
        {
            _entities.AddToProductSet(productToCreate);
            _entities.SaveChanges();
        }

    }

    public interface IProductRepository
    {
        IEnumerable<Product> ListProducts();
        void CreateProduct(Product productToCreate);
    }
}
